### PR TITLE
feat(data-warehouse): Use delta format for DLT

### PR DESCRIPTION
**Pull request by EDsCODE**
## Problem
https://github.com/PostHog/posthog/issues/23648

## Changes
- Use Delta Lake format for all sourced syncs going forwards
- Added conditional logic to do either a `s3` or `deltaLake` function call in clickhouse to read the correct data format
- Added a migration path from Parquet sources to Delta Lake sources
  - We're using the same S3 bucket folder for this
- Known issues:
  - Incremental row merges still don't work yet, but DLT are adding a fix for this
  - Empty tables don't get materialized in S3
- Updates all sources to use Delta format

## Does this work well for both Cloud and self-hosted?
Yup

## How did you test this code?
Tests

### CODE REVIEW

File: posthog/warehouse/models/table.py [hunk 1]
```diff
@@ -61,6 +61,7 @@ class TableFormat(models.TextChoices):
         CSVWithNames = "CSVWithNames", "CSVWithNames"
         Parquet = "Parquet", "Parquet"
         JSON = "JSONEachRow", "JSON"
+        Delta = "Delta", "Delta"
```
Comment (EDsCODE) [inline posthog/warehouse/models/table.py:64]: Clever!

**Review by EDsCODE — COMMENTED** [review]
- I think this is what you mentioned but when i reload or change from type to incremental, the table doesn't update

- Seems to work with no issue. I had a stripe sync on old system, hit reload while on this branch, new data is queryable. I also tested incremental append and it works too

Will do code review next

File: posthog/hogql/database/s3_table.py [hunk 2]
```diff
@@ -1,9 +1,76 @@
...
+        raw_params[param_name] = value
+        return f"%({param_name})s"
+
+    def return_expr(expr: str) -> str:
+        if context is not None:
+            return f"{expr})"
+
+        return f"{substitute_params(expr, raw_params)})"
+
+    if format == "Delta":
```
Comment (EDsCODE) [inline posthog/hogql/database/s3_table.py:35]: 👍

File: posthog/temporal/data_imports/pipelines/pipeline.py [hunk 3]
```diff
@@ -63,12 +64,16 @@ def _get_destination(self):
                 "aws_access_key_id": settings.AIRBYTE_BUCKET_KEY,
                 "aws_secret_access_key": settings.AIRBYTE_BUCKET_SECRET,
                 "endpoint_url": settings.OBJECT_STORAGE_ENDPOINT,
+                "region_name": settings.AIRBYTE_BUCKET_REGION,
+                "AWS_ALLOW_HTTP": "true",
+                "AWS_S3_ALLOW_UNSAFE_RENAME": "true",
```
Comment (EDsCODE) [inline posthog/temporal/data_imports/pipelines/pipeline.py:69]: Bumping to double check: is this the right place for these credentials or do they need to be set on the worker itself too? I didn't get to test this and I have always had this manually set on the env when running the temporal worker locally

File: posthog/temporal/data_imports/pipelines/sql_database/__init__.py [hunk 4]
```diff
@@ -138,6 +138,7 @@ def sql_database(
             merge_key=get_primary_key(table),
             write_disposition="merge" if incremental else "replace",
             spec=SqlDatabaseTableConfiguration,
+            table_format="delta",
```
Comment (EDsCODE) [inline posthog/temporal/data_imports/pipelines/sql_database/__init__.py:141]: Note: did not QA database

**Review by EDsCODE — COMMENTED** [review]
Looks good. I noted a few places to double check that I missed in QA and didn't get to tonight. also, when ready, this is technically my PR so you can give yourself a stamp 😏

File: posthog/temporal/data_imports/pipelines/pipeline.py (hunk 3 above)
Comment (Gilbert09) [reply posthog/temporal/data_imports/pipelines/pipeline.py:69]: These are just for running the local tests. `AWS_ALLOW_HTTP` is needed for delta-rs to talk to local MinIO. They get explicitly passed through to DLT and delta-rs

**Review by Gilbert09 — APPROVED** [review]

**Pull request merged**